src/NuGet.Core/NuGet.Packaging/Signing/Verification/SignatureTrustAndValidityVerificationProvider.cs

```diff
@@ -32,342 +32,44 @@ public Task<PackageVerificationResult> GetTrustResultAsync(ISignedPackageReader
-
-                    issues.Add(SignatureLog.DetailedLog(CertificateUtility.X509ChainToString(chain, _fingerprintAlgorithm)));
-
-                    if (chainBuildSucceed)
-                    {
-                        return true;
-                    }
-
-                    var chainBuildingHasIssues = false;
-                    IEnumerable<string> messages;
-
-                    var timestampInvalidCertificateFlags = CertificateChainUtility.DefaultObservedStatusFlags |
-                        (X509ChainStatusFlags.Revoked) |
-                        (X509ChainStatusFlags.NotTimeValid) |
-                        (X509ChainStatusFlags.CtlNotTimeValid);
-
-                    if (CertificateChainUtility.TryGetStatusMessage(chainStatusList, timestampInvalidCertificateFlags, out messages))
-                    {
-                        foreach (var message in messages)
-                        {
-                            issues.Add(SignatureLog.Issue(treatIssueAsError, NuGetLogCode.NU3028, message));
-                        }
-
-                        chainBuildingHasIssues = true;
-                    }
-
-                    // For all the special cases, chain status list only has unique elements for each chain status flag present
-                    // therefore if we are checking for one specific chain status we can use the first of the returned list
-                    // if we are combining checks for more than one, then we have to use the whole list.
-
-                    const X509ChainStatusFlags RevocationStatusFlags = X509ChainStatusFlags.RevocationStatusUnknown | X509ChainStatusFlags.OfflineRevocation;
-                    if (CertificateChainUtility.TryGetStatusMessage(chainStatusList, RevocationStatusFlags, out messages))
-                    {
-                        if (treatIssueAsError)
-                        {
-                            foreach (var message in messages)
-                            {
-                                issues.Add(SignatureLog.Issue(!allowUnknownRevocation, NuGetLogCode.NU3028, message));
-                            }
-                        }
-
-                        if (!chainBuildingHasIssues && (allowIgnoreTimestamp || allowUnknownRevocation))
-                        {
-                            return true;
-                        }
-
-                        chainBuildingHasIssues = true;
-                    }
-
-                    // Debug log any errors
-                    issues.Add(
-                        SignatureLog.DebugLog(
-                            string.Format(
-                                CultureInfo.CurrentCulture,
-                                Strings.ErrorInvalidCertificateChain,
-                                string.Join(", ", chainStatusList.Select(x => x.Status.ToString())))));
-                }
-            }
+            signatureIssues.AddRange(timestampIssues);
```

Comment:
This is to display first the details of the signature and afterwards for its timestamp (if present). This matches signtool and makes more sense to display first the signature. (example output of signtool https://github.com/nuget/home/issues/6005#issuecomment-349099109)